### PR TITLE
[ASTGen] Generate EditorPlaceholderExpr and 'do' expressions

### PR DESCRIPTION
**Pull request by rintaro**

### CODE REVIEW

File: lib/Parse/ParseDecl.cpp [hunk 1]
```diff
@@ -356,9 +356,11 @@ void Parser::parseSourceFileViaASTGen(
                                        appendToVector);
 
     // Spin the C++ parser to the end; we won't be using it.
+    DiagnosticTransaction noDiag(Context.Diags);
```
Comment (rintaro) [inline lib/Parse/ParseDecl.cpp:359]: This avoids `consumeToken()` emits lexer diagnostics e.g. "editor placeholder in source file"

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Parse/ParseDecl.cpp (hunk 1 above)
Comment (DougGregor) [reply lib/Parse/ParseDecl.cpp:359]: Ah, nice.

**Review by DougGregor — APPROVED** [review]
This all looks good to me!

**Pull request merged**